crates/floresta-cli/src/parsers.rs

```diff
@@ -0,0 +1,60 @@
+use std::any::type_name;
+use std::error::Error;
+use std::fmt::Display;
+use std::str::FromStr;
+
+#[derive(Debug)]
+/// Collection of errors to deal with parsing.
+pub enum ParseError {
+    /// Returned when the user inserts an broken array
+    InvalidHashArray,
+    /// Returned when the consumer of jsonlike_arrayofhashes tries to
+    /// cast into a incompatible type.
+    InvalidHashTarget(String),
+}
+impl Display for ParseError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ParseError::InvalidHashArray => write!(
+                f,
+                "Couldnt parse the inserted as a Hash Array, please refer to the docs"
+            ),
+            ParseError::InvalidHashTarget(target) => {
+                write!(f, "Could parse the hash itens to {target}")
+            }
+        }
+    }
+}
+
+impl Error for ParseError {}
+
+pub type Result<T> = std::result::Result<T, ParseError>;
+/// Tries to parse a json array of hashes256, you can insert a type to be casted on each hash256.
+///
+/// Example: '["4a5e1e4baab89f3a32518a88c31bc87f618f76673e2cc77ab2127b7afdeda33b", "4a5e1e4baab89f3a32518a88c31bc87f618f76673e2cc77ab2127b7afdeda33b"]'
+/// Tries to parse a JSON array of hash strings into a vector of the target type.
+///
+/// The target type must implement Deserialize and can be converted from a hash string.
+/// By default, it will parse into Hash256, but you can specify any other compatible type.
+///
+/// Example:
+/// ```
+/// let hashes: Vec<Hash256> = jsonlike_arrayofhashes(
+///     r#"["4a5e1e4baab89f3a32518a88c31bc87f618f76673e2cc77ab2127b7afdeda33b"]"#,
+/// )?;
+/// ```
+pub fn jsonlike_arrayofhashes<Target>(s: &str) -> Result<Vec<Target>>
```

Comment:
It seems like this method parses more than hashes. How about "parse_json_array"?